std/http/_io_test.ts

```diff
@@ -164,7 +164,7 @@ test("writeTrailer should throw", async () => {
           new Headers({ [f]: "1" })
         );
       },
-      AssertionError,
+      AssertionFailedError,
```

Comment:
I see we've been using assertion errors for validating input 🙃 We have `TypeError` and `InvalidData` for that, this should be fixed.